tool/tsh/tsh.go

```diff
@@ -599,14 +599,27 @@ func Run(args []string, opts ...cliOption) error {
 		// This should only happen when there's a missing switch case above.
 		err = trace.BadParameter("command %q not configured", command)
 	}
+
+	if trace.IsNotImplemented(err) {
+		tc, err := makeClient(&cf)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		pr, err := tc.Ping(ctx)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		return trace.NotImplemented("This server does not implement this feature yet. Likely the client version you are using is newer than the server. The server version: %v, the client version: %v. Please upgrade the server or downgrade the client.", pr.ServerVersion, teleport.Version)
```

Comment:
Let's move this into a helper function and treat errors from makeClient and Ping calls gracefully so we can still return NotImplemented error but use some placeholder for server version if any of those calls failed.

Reply:
done